lib/authorization-builder.js

```diff
@@ -0,0 +1,113 @@
+"use strict";
+
+
+/**
+ * @typedef {Object} AuthorizationBuilder - Builds components for signing requests for OpenToken.
+ * @function buildHeaderValue
+ * @function addSigningContent
+ * @function createHeader
+ * @property {Object} requireOptions
+ */
+
+
+/**
+ * @param {ContentHasher} ContentHasher
+ * @param {log} logger
+ * @param {urijs} Uri
+ * @return {AuthorizationBuilder}
+ */
+module.exports = (ContentHasher, logger, Uri) => {
+    /**
+     * Responsible for building the authorization header for an opentoken request.
+     */
+    class AuthorizationBuilder {
+        /**
+         * @param {string} code
+         * @param {string} secret
+         * @param {requestOptions} requestOptions
+         * @param {string|stream~Readable} body
+         */
+        constructor(code, secret, requestOptions, body) {
+            this.code = code;
+            this.secret = secret;
+            this.body = body;
+            this.requestOptions = requestOptions;
+
+            // This is done here so that when I used it in two different places
```

Comment:
I'm assuming "it" in this case is `this.headerNameList`, but it may be good to reword this so that an assumption isn't necessary.